### PR TITLE
refactor(parse5): parse5 2.* & Refactor ng-core parse5_adapter

### PR DESCRIPTION
**Pull request by MarkPieszak**
Upgraded parse5 to 2.*
Refactored @angular core parse5_adapter locally to use it as well.

Everything works (client gets good response from Universal/etc), but throws server error from `@angular/platform-browser/src/animate/ browser_details.ts`

**Leading to: `TypeError: Cannot read property 'logGroup' of null` showing up on the server**

```
/**
   * Determines if `event.elapsedTime` includes transition delay in the current browser.  At this
   * time, Chrome and Opera seem to be the only browsers that include this.
   */
  doesElapsedTimeIncludesDelay(): void {
    var div = getDOM().createElement('div'); // <-- here (Cannot createElement from null)
    getDOM().setAttribute(div, 'style',
```

**Stacktrace for original uncaught exception error:**
> 
"TypeError: Cannot read property 'createElement' of null\n    at BrowserDetails.doesElapsedTimeIncludesDelay (/Users/Philly/Desktop/GitHub/universal/examples/node_modules/@angular/platform-browser/src/animate/browser_details.js:16:41)\n    at new BrowserDetails (/Users/Philly/Desktop/GitHub/universal/examples/node_modules/@angular/platform-browser/src/animate/browser_details.js:8:14)\n    at ReflectionCapabilities.factory (/Users/Philly/Desktop/GitHub/universal/examples/node_modules/@angular/core/src/reflection/reflection_capabilities.js:12:45)\n    at ReflectiveInjector_._instantiate 
..."

**Stacktrace from server-error that shows up:**

> "TypeError: Cannot read property 'logGroup' of null\n    at ExceptionHandler.call (/Users/Philly/Desktop/GitHub/universal/examples/node_modules/@angular/core/src/facade/exception_handler.js:57:21)\n    at /Users/Philly/Desktop/GitHub/universal/examples/node_modules/@angular/core/src/application_ref.js:301:41\n    at ZoneDelegate.invoke (/Users/Philly/Desktop/GitHub/universal/examples/node_modules/zone.js/dist/zone-node.js:281:29)\n    at Object.NgZoneImpl.inner.inner.fork.onInvoke (/Users/Philly/Desktop/GitHub/universal/examples/node_modules/@angular/core/src/zone/ng_zone_impl.js:45:41)\n    at ZoneDelegate.invoke (/Users/Philly/Desktop/GitHub/universal/examples/node_modules/zone.js/dist/zone-node.js:280:35)\n    at Zone.run (/Users/Philly/Desktop/GitHub/universal/examples/node_modules/zone.js/dist/zone-node.js:174:44)\n    at NgZoneImpl.runInner (/Users/Philly/Desktop/GitHub/universal/examples/node_modules/@angular/core/src/zone/ng_zone_impl.js:76:71)\n    at NgZone.run (/Users/Philly/Desktop/GitHub/universal/examples/node_modules/@angular/core/src/zone/ng_zone.js:223:66)\n    at ApplicationRef_.run (/Users/Philly/Desktop/GitHub/universal/examples/node_modules/@angular/core/src/application_ref.js:290:14)\n    at ApplicationRef_.bootstrap (/Users/Philly/Desktop/GitHub/universal/examples/node_modules/@angular/core/src/application_ref.js:312:21)"

### CODE REVIEW

**Comment by gdi2290** [comment]
@MarkPieszak I'm looking to merge this. can you give me an update whenever you can? I'll work with the old version for now

**Comment by gdi2290** [comment]
thanks, this is in master now

File: modules/universal/src/node/platform/dom/parse5_adapter.ts [hunk 1]
```diff
@@ -0,0 +1,818 @@
...
+    /* MarkPieszak - replace parse5_adapter
+    parser = new parse5.Parser(parse5.TreeAdapters.htmlparser2);
+    serializer = new parse5.Serializer(parse5.TreeAdapters.htmlparser2);
+    treeAdapter = parser.treeAdapter;
+    setRootDomAdapter(new Parse5DomAdapter());
+    */
+
+    parser = parse5;
+    serializer = parse5.serialize;
+    treeAdapter = parse5.treeAdapters.htmlparse2;
```
Comment (gdi2290) [inline modules/universal/src/node/platform/dom/parse5_adapter.ts]: there was a typo here that was hard to track
should have been `htmlparser2`

Comment (MarkPieszak) [inline modules/universal/src/node/platform/dom/parse5_adapter.ts]: Oh wow nice catch!
Is this merged in now? Nice!! @gdi2290

**Pull request closed without merging**